docs/sovereign/standalone-evm.md

```diff
@@ -0,0 +1,172 @@
+
+EVM gas metering is handled internally within the EVM code. The VMExecutionHandler can receive a new gas schedule via:
+
+```go
+GasScheduleChange(newGasSchedule map[string]map[string]uint64)
+```
+
+This function provides the cost of each opcode as a map. The EVM needs the appropriate wrapper functions to load these costs into its **OPCODES** structure.
+
+---
+
+## 5. Implementation Steps: Integrating EVM
+
+- Start from the SpaceVM code.
+- Replace the current executor (WASMER) with the EVM executor.
+- During EVM opcode interpretation, invoke the storageContext and meteringContext functions to manage state changes and track gas consumption.
+
+Once these steps are complete, the underlying EVM logic should effectively run on MultiversX.
+
+---
+
+## 6. Address Conversion: 20 Bytes vs. 32 Bytes
+
+EVM addresses are 20-bytes long, whereas MultiversX uses 32-byte addresses. To avoid changing the broader MultiversX system, the EVM will use internal transformers:
+
+- **Internal EVM Calls**: Within the EVM, contracts use the last 20 bytes of the corresponding 32-byte MvX address.
+- At runtime, the full 32-byte address is still known, and when a storage `read` or `write` occurs, the EVM prefixes the last 20 bytes with 10 bytes of zeros plus a 2-byte `VMType` (the standard MvX smart contract addressing scheme).
+
+### 6.1 Calling EVM Contracts from EVM
+
+When an EVM-based smart contract calls another EVM contract, it uses the 20-byte address. Internally, the system prefixes these 20 bytes with the deterministic overhead (10 bytes of zeros and 2 bytes for `VMType`) to fetch the appropriate contract code from the accounts trie before running it.
+
+### 6.2 Token Storage in EVM
+
+Token balances (like ERC20) live in the contract’s own storage. The contract will use the last 20 bytes of a user’s MvX address when recording ownership or balances. If an opcode like `GetCaller` is executed, it returns only the last 20 bytes from the `ContractCallInput.Sender`.
+
+### 6.3 Calling WasmVM from EVM
+
+MultiversX WasmVM expects 32-byte addresses. If an EVM contract tries to invoke a WasmVM contract using only 20 bytes, the call will fail due to incorrect argument size. Consequently, when the EVM calls a WasmVM contract, it must supply a full 32-byte address.
+
+:::note
+In most cases, the EVM contracts will call only other EVM contracts. However, bridging to WasmVM is still feasible, for example, when claiming ESDT tokens through an ERC wrapper contract.
+:::
+
+---
+
+## 7. WASM VM and the `ExecuteOnDestOnOtherVM` Function
+
+The **WASM VM** supports a public function `ExecuteOnDestOnOtherVM` via the **BlockchainHook** interface. If a new VM is fully integrated, it can be added to the `vmContainer` component with a new **baseAddress**. Below is an example table illustrating potential base addresses for different VMs:
+
+| VM Name     | Example Base Address | Notes                                                                            |
```

Comment:
Maybe "VM Type" instead of "base address"? Or "address suffix"? (just an opinion)